controllers/space/space_controller_test.go

```diff
@@ -1029,6 +1031,127 @@ func TestUpdateSpaceTier(t *testing.T) {
+		member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
+		member2 := NewMemberCluster(t, "member-2", corev1.ConditionTrue)
+
+		ctrl := newReconciler(hostClient, member1, member2)
+
+		// when
+		res, err := ctrl.Reconcile(context.TODO(), requestFor(s))
+
+		// then
+		require.NoError(t, err)
+		assert.True(t, res.Requeue) // expect a requeue since the NSTemplateSet was updated
+		// Space should be in "updating" state while the
+		spacetest.AssertThatSpace(t, test.HostOperatorNs, s.Name, hostClient).
+			HasConditions(spacetest.Updating())
+		// NSTemplateSet should have an spaceRoles entry for the `mur`
+		nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, nstmplSet.Name, member1Client).
+			HasSpaceRoleForUser("basic-admin-123456new", mur.Name). // entry added for user
+			HasConditions(nstemplatetsettest.Provisioned())         // not changed by the SpaceController, but will be by the NSTemplateSetController
+	})
+
+	t.Run("remove user with admin role", func(t *testing.T) {
+		// given a MUR, a Space and its NSTemplateSet resource...
+		mur := murtest.NewMasterUserRecord(t, "john")
+		s := spacetest.NewSpace("oddity",
+			spacetest.WithTierName(basicTier.Name),
+			spacetest.WithSpecTargetCluster("member-1"),
+			spacetest.WithStatusTargetCluster("member-1"), // already provisioned on a target cluster
+			spacetest.WithFinalizer())
+		nstmplSet := nstemplatetsettest.NewNSTemplateSet("oddity",
+			nstemplatetsettest.WithReferencesFor(basicTier,
+				nstemplatetsettest.WithSpaceRole("admin", mur.Name),
+			),
+			nstemplatetsettest.WithReadyCondition())
+		// but no SpaceBinding anymore for John as an admin on the Space
+		hostClient := test.NewFakeClient(t, s, mur, basicTier)
+		member1Client := test.NewFakeClient(t, nstmplSet)
+		member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
+		member2 := NewMemberCluster(t, "member-2", corev1.ConditionTrue)
+
+		ctrl := newReconciler(hostClient, member1, member2)
+
+		// when
+		res, err := ctrl.Reconcile(context.TODO(), requestFor(s))
+
+		// then
+		require.NoError(t, err)
+		assert.True(t, res.Requeue) // expect a requeue since the NSTemplateSet was updated
+		// Space should be in "updating" state while the
+		spacetest.AssertThatSpace(t, test.HostOperatorNs, s.Name, hostClient).
+			HasConditions(spacetest.Updating())
+		// NSTemplateSet should have an spaceRoles entry for the `mur`
+		nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, nstmplSet.Name, member1Client).
+			HasNoSpaceRoleForUser("basic-viewer-123456new", mur.Name). // no change here
+			HasNoSpaceRoleForUser("basic-admin-123456new", mur.Name).  // entry removed for user
+			HasConditions(nstemplatetsettest.Provisioned())            // not changed by the SpaceController, but will be by the NSTemplateSetController
+	})
+
+	t.Run("update user from viewer to admin role", func(t *testing.T) {
```

Comment:
analogically the same here - could we add more SpaceRoles/SpaceBindings as "noise" and verify that the controller updated only one?

Reply:
done in 9d23775